browser/onboarding/onboarding_unittest.cc

```diff
@@ -43,11 +43,27 @@ class OnboardingTest : public testing::Test {
     profile_manager_->DeleteTestingProfile(kTestProfileName);
   }
 
+  void TabHelperCreationCallback() {
+    tab_helper_creation_callback_called_ = true;
+  }
+
+  void MaybeCreateForWebContentsAndWaitTillGetResult(
+      content::WebContents* web_contents) {
+    tab_helper_creation_callback_called_ = false;
+    web_contents->SetUserData(OnboardingTabHelper::UserDataKey(), nullptr);
+    OnboardingTabHelper::MaybeCreateForWebContents(
+        web_contents, base::BindOnce(&OnboardingTest::TabHelperCreationCallback,
+                                     base::Unretained(this)));
+    ASSERT_TRUE(base::test::RunUntil(
+        [this] { return tab_helper_creation_callback_called_; }));
+  }
```

Comment:
I particularly would do something like

```cxx
  void MaybeCreateForWebContentsAndWaitTillGetResult(
      content::WebContents* web_contents) {
    bool created = false;
    web_contents->SetUserData(OnboardingTabHelper::UserDataKey(), nullptr);
    OnboardingTabHelper::MaybeCreateForWebContents(
        web_contents, base::BindLambdaForTesting([&created]{
            created = true;
     }));
    ASSERT_TRUE(base::test::RunUntil(
        [&created] { return created; }));
  }
```

So in this case we just use the local variable, and avoid `base::Unretained` and having to have a member variable to the test.

Reply:
Much better. Updated!